src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -572,6 +572,40 @@ trait Implicits {
        }
      }
 
+    private def matchesPtInst(info: ImplicitInfo): Boolean = {
+      // the pt for views can have embedded unification type variables or BoundedWildcardTypes which
+      // can't be solved for. Rather than attempt to patch things up later we just skip those cases
+      // altogether.
+      def isInstantiable =
+        !isView || (wildPt.isGround && !wildPt.exists { case _: BoundedWildcardType => true ; case _ => false })
+
+      info.tpe match {
+        case PolyType(tparams, restpe) if isInstantiable =>
+          val tp = ApproximateDependentMap(restpe)
+          val allUndetparams = (undetParams ++ tparams).distinct
+          try {
+            val tvars = allUndetparams map freshVar
+            val tpInstantiated = tp.instantiateTypeParams(allUndetparams, tvars)
+            if(!matchesPt(tpInstantiated, wildPt, allUndetparams)) false
```

Comment:
I'd like to see some statistics of which branch prunes the most in a few projects. Maybe we don't even need to solve types to still get some nice pruning? Could we first (1) check whether `tp.instantiateTypeParams(allUndetparams, allUndetparams map (_ => WildcardType))` matches `wildPt`, then (2) check with type vars, and (3) finally solve?

Would like to see hit counts for branches (1), (2), (3).